### PR TITLE
Add disable build server CLI option

### LINKED ISSUE #10288: Implement an option for restore/build/publish commands that disable build servers.
## Steps to reproduce

See discussion on https://github.com/dotnet/cli/issues/9481.

1) `dotnet new console`
2) `dotnet build`

## Expected  behavior

There should be an option to `dotnet build` (and other building commands) to disable the use of the various build servers (Roslyn, MSBuild node reuse, Razor).

## Actual behavior

Users have to pass a plethora of undiscoverable properties and options to fully disable the use of the build servers, which might be helpful for CI scenarios.

## Environment data

```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview5-011568
 Commit:    b487ff10aa

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.0.100-preview5-011568/

Host (useful for support):
  Version: 3.0.0-preview5-27626-15
  Commit:  61f30f5a23

.NET Core SDKs installed:
  3.0.100-preview5-011568 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0-preview5-19227-01 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0-preview5-27626-15 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

### PR DESCRIPTION
**Pull request by sfoslund**
Fixes https://github.com/dotnet/sdk/issues/10288

### CODE REVIEW

**Comment by sfoslund** [comment]
@baronfel can I get you to review the option name and help text here?

**Review by joeloff — APPROVED** [review]

File: src/Cli/dotnet/CommonLocalizableStrings.resx [hunk 1]
```diff
@@ -699,4 +699,7 @@ The default is 'true' if a runtime identifier is specified.</value>
   <data name="SelfContainAndNoSelfContainedConflict" xml:space="preserve">
     <value>The '--self-contained' and '--no-self-contained' options cannot be used together.</value>
   </data>
+  <data name="DisableBuildServersOptionDescription" xml:space="preserve">
+    <value>Disable the use of build servers.</value>
```
Comment (baronfel) [inline src/Cli/dotnet/CommonLocalizableStrings.resx:703]:
```suggestion
    <value>Force the command to ignore any persistent build servers.</value>
```

what do you think of this phrasing @sfoslund? I'm patterning it mostly off of the phrasing for `CommandInteractiveOptionDescription` on line 671, aiming for consistent tone overall.

**Review by dsplaisted — APPROVED** [review]
Looks good.

Did you do a sanity test to verify manually that it doesn't use the build servers (but should still do a multi-proc build if there are multiple projects)?

File: src/Cli/dotnet/CommonLocalizableStrings.resx (hunk 1 above)
Comment (sfoslund) [reply src/Cli/dotnet/CommonLocalizableStrings.resx]: Makes sense to me, just updated to that string.

**Comment by sfoslund** [comment]
> Did you do a sanity test to verify manually that it doesn't use the build servers (but should still do a multi-proc build if there are multiple projects)?

I checked that the properties flow correctly, I'm not sure how to do further tests than that. Do you have any suggestions?

**Comment by dsplaisted** [comment]
> > Did you do a sanity test to verify manually that it doesn't use the build servers (but should still do a multi-proc build if there are multiple projects)?
> 
> I checked that the properties flow correctly, I'm not sure how to do further tests than that. Do you have any suggestions?

In PowerShell, this will list the command lines of currently running `dotnet.exe` processes:

```powershell
 gcim win32_process | Where-Object {$_.Name -eq "dotnet.exe"} | select CommandLine | Format-table -autosize -Wrap
```

MSBuild nodes will have MSBuild.dll on the command line, and the Roslyn compiler server will have VBCSCompiler.dll.  I'm not sure what the Razor server uses and you would need a Razor project to trigger it.

So I'd do something like the following:

- `taskkill /f /im dotnet.exe` to kill all dotnet processes
- `dotnet build` on a project with multiple project references (so that it starts multiple MSBuild nodes)
- Verify that there are MSBuild.dll and a VBCSCompiler.dll processes
- Kill all dotnet processes again
- `dotnet build -disable-build-servers`
- Verify that there are no remaining dotnet processes after the build completes

**Comment by sfoslund** [comment]
@dsplaisted awesome, just verified that things are working as you described.

**Pull request merged**